src/map/skill.c

```diff
@@ -15623,20 +15623,49 @@ bool skill_check_condition_castbegin(struct map_session_data* sd, uint16 skill_i
 		}
 	}
 
-	//check if equipped item
+	// Check for equipped item(s)
 	if (require.eqItem_count) {
+		uint8 count = require.eqItem_count;
+
 		for (i = 0; i < require.eqItem_count; i++) {
 			uint16 reqeqit = require.eqItem[i];
 
 			if (!reqeqit)
-				break; //no required item; get out of here
-			if (!pc_checkequip2(sd,reqeqit,EQI_ACC_L,EQI_MAX)) {
-				if (i == require.eqItem_count) {
-					clif_skill_fail(sd,skill_id,USESKILL_FAIL_THIS_WEAPON,0);
-					return false;
-				}
-			} else
-				break; // Wearing an applicable item.
+				break; // Skill has no required item(s); get out of here
+			switch(skill_id) { // Specific skills require multiple items while default will handle singular cases
+				case NC_PILEBUNKER:
+					if (!pc_checkequip2(sd,reqeqit,EQI_ACC_L,EQI_MAX)) {
+						count--;
+						if (!count) {
+							clif_skill_fail(sd,skill_id,USESKILL_FAIL_THIS_WEAPON,0);
+							return false;
+						} else
+							continue;
+					}
+					break;
+				case NC_ACCELERATION:
+				case NC_SELFDESTRUCTION:
+				case NC_SHAPESHIFT:
+				case NC_EMERGENCYCOOL:
+				case NC_MAGNETICFIELD:
+				case NC_NEUTRALBARRIER:
+				case NC_STEALTHFIELD:
+					if (pc_search_inventory(sd,reqeqit) < 0) {
+						count--;
+						if (!count) {
+							clif_skill_fail(sd,skill_id,USESKILL_FAIL_NEED_EQUIPMENT,reqeqit<<16);
```

Comment:
Probaby this should use:
`clif_skill_fail(sd,skill_id,USESKILL_FAIL_NEED_ITEM,reqeqit<<16|1);`